### PR TITLE
ARROW-332: Add RecordBatch.to_pandas method

### PR DESCRIPTION
**Pull request by wesm**
This makes testing and IPC data wrangling a little easier.

### CODE REVIEW

**Comment by xhochy** [comment]
@wesm Can you rebase? With the ARROW-312 changes inside, it's hard to review.

**Comment by wesm** [comment]
Sorry about that, somehow I got my master branch into a bad state, but OK now

File: python/pyarrow/tests/test_table.py [hunk 1]
```diff
@@ -15,28 +15,47 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import pyarrow as A
+import numpy as np
+
+from pandas.util.testing import assert_frame_equal
+import pandas as pd
+
+import pyarrow as pa
```
Comment (xhochy) [inline python/pyarrow/tests/test_table.py]: Nitpick: We should settle on a single abbreviation for `pyarrow`

**Review by xhochy — APPROVED** [review]

**Comment by wesm** [comment]
+1

**Pull request closed without merging**